projects/fsastorefrontlib/src/cms-components/agent/agent-search-list/agent-search-list.component.html

```diff
@@ -76,26 +76,51 @@
                 'd-none d-lg-flex': !initialActiveAgent && selectedIndex !== i
               }"
             >
-              <div class="col-12 col-sm-3 position-relative text-center mb-3">
-                <div class="counter text-center semi-bold position-absolute">
-                  {{ i + 1 }}
-                </div>
-                <cx-media
-                  class="cx-product-image-container px-4"
-                  [container]="agent.thumbnail"
-                  format="thumbnail"
-                >
-                </cx-media>
-                <ng-container *ngIf="agent.participationStatus; else offline">
-                  <div class="mt-3 notice">
-                    <span class="mr-1 fas fa-comment"></span>
-                    <span>
-                      {{ 'fscommon.agentLocator.online' | cxTranslate }}
-                    </span>
+              <div class="col-12 col-sm-4">
+                <div class="d-flex justify-content-center">
+                  <div class="counter text-center semi-bold position-absolute">
+                    {{ i + 1 }}
+                  </div>
+                  <div class="position-relative">
+                    <cx-media
+                      [ngbTooltip]="
+                        agent.participationStatus
+                          ? ('fscommon.agentLocator.online' | cxTranslate)
+                          : ('fscommon.agentLocator.offline' | cxTranslate)
+                      "
+                      [disableTooltip]="!(user$ | async)"
+                      placement="bottom-right"
+                      class="cx-product-image-container m-auto"
+                      [container]="agent.thumbnail"
+                      format="thumbnail"
+                    >
+                    </cx-media>
+                    <span
+                      [class]="agent.participationStatus ? 'online' : 'offline'"
+                      *ngIf="user$ | async"
+                    ></span>
                   </div>
-                </ng-container>
+                </div>
+                <div class="d-flex mt-3 mb-3 mb-md-0 justify-content-center">
+                  <a
+                    class="fas fa-envelope"
+                    [class.pr-2]="agent.participationStatus && (user$ | async)"
```

Comment:
`user$ | async` should be hoisted somewhere on top, so not to repeat it.

Reply:
If I move it somewhere up, when the user is logged out, won't see agents or some other content.

Reply:
I could only unpack it in the component?

Reply:
I would try several approaches here.
The easiest one would be to transform this observable with `shareReplay()` or `share()` RxJS operator. That would maybe be the quick and dirty way, but it would ensure you are not calling subscription each time `async` pipe is used.
Other approach (cleaner, easier to follow and maintain) would be to try to separate structure from content rendering using Angular's template ref and `ng-container` as described in this [example](https://stackoverflow.com/a/69442257)

Reply:
ng-template won't solve the problem, because I also need a user outside the ng-template. I'll try with the first suggestion.